pkg/prompt/stubber.go

```diff
@@ -2,104 +2,204 @@ package prompt
 
 import (
 	"fmt"
-	"reflect"
+	"strings"
 
 	"github.com/AlecAivazis/survey/v2"
 	"github.com/AlecAivazis/survey/v2/core"
+	"github.com/cli/cli/v2/pkg/surveyext"
 )
 
 type AskStubber struct {
-	Asks     [][]*survey.Question
-	AskOnes  []*survey.Prompt
-	Count    int
-	OneCount int
-	Stubs    [][]*QuestionStub
-	StubOnes []*PromptStub
+	stubs []*QuestionStub
 }
 
-func InitAskStubber() (*AskStubber, func()) {
+type testing interface {
+	Errorf(format string, args ...interface{})
+}
+
+func NewAskStubber() (*AskStubber, func(t testing)) {
```

Comment:
What are your thoughts on passing `t *testing.T` into `NewAskStubber` and automatically registering the teardown function using `t.Cleanup`?

```golang
func NewAskStubber(t *testing.T) *AskStubber {
```

This would simplify the use of the stubber and make sure we don't run into the case of forgetting to do the teardown causing hard to diagnose issues elsewhere in the test suite.

Reply:
That's a great idea! I'll switch to that approach 👍